src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+        tmpCopy.setDissectSpiroCarbonSetting(this.dissectSpiroCarbonSetting.get());
+        tmpCopy.setDissectRingsSetting(this.dissectRingsSetting.get());
+        tmpCopy.setPreserveRingMaxSizeSetting(this.preserveRingMaxSizeSetting.get());
+        tmpCopy.setPreserveRingSystemSetting(this.preserveRingSystemSetting.get());
+        tmpCopy.setPreserveRingSystemMaxSetting(this.preserveRingSystemMaxSetting.get());
+        return tmpCopy;
+    }
+    /**
+     * Restore all settings of the fragmenter to their default values.
+     */
+    @Override
+    public void restoreDefaultSettings() {
+        this.fragmentSaturationSetting.set(IMoleculeFragmenter.FRAGMENT_SATURATION_OPTION_DEFAULT.name());
+        this.chainFragmentLengthSetting.set(AlkylStructureFragmenter.Chain_Fragment_LENGTH_OPTION_DEFAULT.name());
+        this.dissectSpiroCarbonSetting.set(AlkylStructureFragmenter.DISSECT_SPIRO_CARBON_OPTION_DEFAULT);
+        this.dissectRingsSetting.set(AlkylStructureFragmenter.DISSECT_RINGS_OPTION_DEFAULT);
+        this.preserveRingMaxSizeSetting.set(AlkylStructureFragmenter.Preserve_Ring_MAX_SIZE_OPTION_DEFAULT.name());
+        this.preserveRingSystemSetting.set(AlkylStructureFragmenter.PRESERVE_RING_SYSTEM_OPTION_DEFAULT);
+        this.preserveRingSystemMaxSetting.set(AlkylStructureFragmenter.PRESERVE_RING_SYSTEM_MAX_OPTION_DEFAULT);
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Fragmentation">
+    /**
+     * Fragments a clone(!) of the given molecule according to the respective algorithm and returns the resulting fragments.
+     *
+     * @param aMolecule to fragment
+     * @return a list of fragments (the list may be empty if no fragments are extracted, but the fragments should not be!)
+     * @throws NullPointerException       if aMolecule is null
+     * @throws IllegalArgumentException   if the given molecule cannot be fragmented but should be filtered or preprocessed
+     * @throws CloneNotSupportedException if cloning the given molecule fails
+     */
+    @Override
+    public List<IAtomContainer> fragmentMolecule(IAtomContainer aMolecule)
+            throws NullPointerException, IllegalArgumentException, CloneNotSupportedException {
+        //
+        //<editor-fold desc="Parameter Checks" defaultstate="collapsed">
+        Objects.requireNonNull(aMolecule, "Given molecule is null.");
+        //</editor-fold>
+        //
+        //<editor-fold desc="Set Property and Arrays" defaultstate="collapsed">
+        // Key: IndexProperty Value = IndexInt; Value: IAtom/IBond with IndexInt
+        IAtom[] tmpAtomArray = new IAtom[aMolecule.getAtomCount()];
+        IBond[] tmpBondArray = new IBond[aMolecule.getBondCount()];
+        int tmpAlkylSFAtomIndexInt = 0;
+        int tmpAlkylSFBondIndexInt = 0;
+        String tmpAlkylSFAtomIndexProperty = "AlkylSFAtomIndex";
+        String tmpAlkylSFBondIndexProperty = "AlkylSFBondIndex";
```

Comment:
In the names, -key should be added. And in the key, I would separate the class name from the actual key phrase with ".". Furthermore, these strings should be defined as public static final class constants.